rslib/src/search/parser.rs

```diff
@@ -359,19 +367,31 @@ fn parse_rated(val: &str) -> ParseResult<SearchNode<'static>> {
 
     let ease = match it.next() {
         Some(v) => {
-            let n: u8 = v.parse()?;
-            if n < 5 {
-                Some(n)
+            let u: u8 = v.parse()?;
+            if (1..5).contains(&u) {
+                EaseKind::AnswerButton(u)
             } else {
                 return Err(ParseError {});
             }
         }
-        None => None,
+        None => EaseKind::AnyAnswerButton,
     };
 
     Ok(SearchNode::Rated { days, ease })
 }
 
+/// eg resched:3
+fn parse_resched(val: &str) -> ParseResult<SearchNode<'static>> {
+    let mut it = val.splitn(1, ':');
```

Comment:
I'm currently resolving the merge conflicts in #922 and was wondering: Is this line intentional?
I think when a user inputs `resched:1:1` he must have very flawed assumptions about this kew word and should get an error message rather than silently stripping away the suffix. Or am I the one with the flawed assumptions?

Reply:
When I currently enter `resched:1:1`, I do get an error however?

Reply:
Ah well, you pass 1, so it's a redundant operation anyway. I have omitted it in my PR.
```
assert_eq!("any string", "any string".splitn(1, ' ').next().unwrap());
```

Reply:
Oh, now I see it! Yes, thanks for omitting it.